src/main/java/io/cdap/plugin/http/source/common/http/OAuthUtil.java

```diff
@@ -54,5 +67,52 @@ public static String getAccessTokenByRefreshToken(CloseableHttpClient httpclient
     JsonElement jsonElement = JSONUtil.toJsonObject(responseString).get("access_token");
     return jsonElement.getAsString();
   }
+
+  public static String getAccessTokenByServiceAccount(BaseHttpSourceConfig config) throws IOException {
+    GoogleCredentials credential;
+    String jwtToken = "";
+    if (config.isServiceAccountJson()) {
+      InputStream jsonInputStream = new ByteArrayInputStream(config.getServiceAccountJson().getBytes());
```

Comment:
Ensure you pass UTF-8 otherwise it's current encoding-dependent. Also it's a good habit to put any InputStreams into try with resources block to ensure it's closed.

Reply:
The function itself throws IOException so I did not add the try block

Reply:
Not try-catch, but try-with-resources